packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts

```diff
@@ -0,0 +1,69 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+/**
+ * Regression test for #14001
+ * Invalid character in mongo query
+ */
+testMatrix.setupTestSuite(
+  () => {
+    test('findFirst', async () => {
+      const result = prisma.resource.findFirst({
+        where: { OrderBy: { gt: 0 } },
+        orderBy: { OrderBy: 'asc' },
+        cursor: { OrderBy: 1 },
+        distinct: ['OrderBy'],
+      })
+
+      await expect(result).resolves.toMatchInlineSnapshot(`null`)
+    })
+
+    test('findMany', async () => {
+      const result = prisma.resource.findMany({
+        where: { OrderBy: { gt: 0 } },
+        orderBy: { OrderBy: 'asc' },
+        cursor: { OrderBy: 1 },
+        distinct: ['OrderBy'],
+      })
+
+      await expect(result).resolves.toMatchInlineSnapshot(`Array []`)
```

Comment:
Wondering if we should be asserting that, (1) the query doesn't crash, (2) that the data returned is correct...

Thinking we should add some dummy data in the beforeAll hook and assert the response of these queries.

Reply:
Yes, I think having dummy data does not hurt, I'll add.

Reply:
Turns out it uncovered something surprising. I will merge PR as-is, and report the crash in a new issue.

Reply:
I got confused, this is the one that I tried to add dummy data to and it crashed https://github.com/prisma/prisma/pull/15215 I reported the issue already